SalesforceFoundation/NPSP
# W-7849631: Process RDs in BDI (3a of 5) - Update install script

[W-7849631](https://gus.my.salesforce.com/a07B0000008WOQaIAO): Process RDs in BDI (3a of 5) - Update NPSP install script to push new object and field mappings

The code (aka installation script), creates migrated recurring_donation object and/or migrated fields dynamically only when they don't exist AND when all the original mapping references exist.  

Notes: https://salesforce.quip.com/wjGWAPKE6f0d

Only 4 classes should appear in Files changed tab:
- BDI_MigrationMappingUtility.cls
- BDI_MigrationMappingUtility_TEST.cls
- STG_InstallScript.cls
- STG_InstallScript_TEST.cls

How to Test this using Dev Console:
--
- Execute the following script in order to upsert migrated Recurring Donation object and migrated Recurring Donation fields.
  - use dev and dev_namespaced orgs 
- You can test deleting different fields mappings and executing the script, should create only the missing mappings.
- If all the migrated objects and fields are present, when running this script it should not do nothing.
- You can also test deleting all migrated field mappings for the migrated recurring-donation object so then you can delete the migrated object mapping and recreate the entire migrated-structure using the script:
```
List<String> newDefaultMappingStrings = new List<String>{          
    'Recurring_Donation_Amount',
    'Recurring_Donation_Date_Established',
    'Recurring_Donation_Day_of_Month',
    'Recurring_Donation_Effective_Date',
    'Recurring_Donation_Elevate_Recurring_Id',
    'Recurring_Donation_End_Date',
    'Recurring_Donation_Installment_Frequency',
    'Recurring_Donation_Installment_Period',
    'Recurring_Donation_Name',
    'Recurring_Donation_Payment_Method',
    'Recurring_Donation_Planned_Installments',
    'Recurring_Donation_Recurring_Type',
    'Recurring_Donation_Status',
    'Recurring_Donation_Status_Reason',
    'Recurring_Donation_Campaign'
};

BDI_MigrationMappingHelper helper = new BDI_MigrationMappingHelper();
BDI_MigrationMappingUtility util = new BDI_MigrationMappingUtility(helper);
util.migrateNewDefaultToCustomMetadata(newDefaultMappingStrings);
```


# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata
